### DISCUSSION TITLE
How do I make a crx / xpi file with wxt?

### DISCUSSION THREAD

**Discussion by ferenc-a** [wxt-dev/wxt#1205] [Q&A] [answered]
I used `npx wxt@latest init`, and checked the docs, but couldn't find information about producing the final extension artifacts. `wxt zip` is not enough, but `wxt submit` is too much for my needs.

**Comment by aklinker1** [comment, marked as answer]
Only the stores can create valid XPI/CRX files, since they are signed zip files, and the browsers no longer support self-signed extensions.

So once you publish a version, you can download the files from the developer dashboards or using the API.

That means you have to publish manually or use `wxt submit` before you can download the signed artifacts.

**Comment by ferenc-a** [comment]
There are still ways to make browsers accept self-signed extensions, or even unzipped extensions. It would be nice to get some support for that, like a way to sign the zip / make it into the crx format.
A related issue I found: https://github.com/wxt-dev/wxt/issues/1730
This also seems related: https://github.com/wxt-dev/wxt/issues/962

**Reply by aklinker1** [reply]
Yes, you can create unsigned archives like so (exact commands might not work without additional flags):

```sh
pnpm wxt build -b chrome
/path/to/chrome --pack-extension=".output/chrome-mv3"

pnpm wxt build -b firefox
pnpm i web-ext -D
pnpm web-ext sign .output/firefox-mv2
```

While it is possible to install both these, it requires turning off security features in the browser, which I don't want to promote. So I won't be adding any sort of support for this.

You can already install ZIP files WXT produces in all browsers without disabling security.
